pom.xml

```diff
@@ -41,499 +45,115 @@ Brotli, Zstandard and ar, cpio, jar, tar, zip, dump, 7z, arj.
-      <artifactId>mockito-junit-jupiter</artifactId>
-      <version>${mockito.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>com.github.marschall</groupId>
-      <artifactId>memoryfilesystem</artifactId>
-      <version>2.8.0</version>
-      <scope>test</scope>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-compress-brotli</artifactId>
+      <version>${project.version}</version>
     </dependency>
 
-    <!-- integration test verifying OSGi bundle works -->
-    <dependency>
-      <groupId>org.ops4j.pax.exam</groupId>
-      <artifactId>pax-exam-container-native</artifactId>
-      <version>${pax.exam.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.ops4j.pax.exam</groupId>
-      <artifactId>pax-exam-junit4</artifactId>
-      <version>${pax.exam.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.ops4j.pax.exam</groupId>
-      <artifactId>pax-exam-cm</artifactId>
-      <version>${pax.exam.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.ops4j.pax.exam</groupId>
-      <artifactId>pax-exam-link-mvn</artifactId>
-      <version>${pax.exam.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.felix</groupId>
-      <artifactId>org.apache.felix.framework</artifactId>
-      <version>7.0.5</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>javax.inject</groupId>
-      <artifactId>javax.inject</artifactId>
-      <version>1</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-api</artifactId>
-      <version>${slf4j.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>commons-io</groupId>
-      <artifactId>commons-io</artifactId>
-      <version>2.15.1</version>
-    </dependency>    
     <dependency>
       <groupId>org.apache.commons</groupId>
-      <artifactId>commons-lang3</artifactId>
-      <version>3.14.0</version>
-    </dependency>    
-    <dependency>
-      <groupId>org.osgi</groupId>
-      <artifactId>org.osgi.core</artifactId>
-      <version>6.0.0</version>
-      <scope>provided</scope>
+      <artifactId>commons-compress-core</artifactId>
+      <version>${project.version}</version>
     </dependency>
-  </dependencies>
 
-  <scm>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-compress-core</artifactId>
+        <version>${project.version}</version>
+        <type>test-jar</type>
+      </dependency>
+
+    </dependencies>
+  </dependencyManagement>
+
+  <modules>
+    <module>compress</module>
+    <module>compress-brotli</module>
+    <module>compress-core</module>
+    <module>compress-parent</module>
```

Comment:
can't we simply have this pom as parent which looks more obvious in the tree :)

Reply:
Do you mean moving `compress-parent/pom.xml` here and this file to `compress-bom/pom.xml`, without changing the inheritance tree (`commons-compress-bom` -> `commons-compress-parent` -> `commons-compress-core`? It should probably work.

In Log4j we structure our projects this way, which probably has to do with the fact that we use `${revision}` as version number and the definition of the `revision` property is in the root `pom.xml`. @vy, are the other reasons to put the BOM at the root of the project?

Reply:
Do you mean why not having a structure as follows?

1. `/commons-bom/pom.xml` (`commons-bom` module)
2. `/pom.xml` (`commons-parent` module parented by `commons-bom`)
3. `/commons-core/pom.xml` (`commons-core` module` parented by `commons-parent`)

This setup has the following shortcoming:

Imagine you have a, say, `maven-enforcer-plugin` configuration that applies to all `pom.xml`s, including `/commons-bom/pom.xml`. If you run `./mvnw enforcer:enforce` at the project root directory, it will only run on `commons-parent` and modules defined in `/pom.xml`. That is, it won't run for `/commons-bom/pom.xml`. You can add more examples: `apache-rat-plugin` won't check the license of `/commons-bom/pom.xml`, etc.